### ISSUE TITLE
Simplify generated Immutable files to support Jackson Builder pattern

### ISSUE THREAD

**Issue by tetv** [immutables/immutables#412]
Suggestion for when it's specified `@JsonDeserialize(builder = MyBuilder.class)` the generated files should be cleaned from all the `@JsonXXX` annotations and `@Deprecated` conversion classes because Jackson by design supports Builder pattern.

**Comment by elucash** [comment]
Yep, makes sense to not interfere with functionality when you want custom binding. Thank you for feature request!

**Comment by elucash** [comment]
when `builder` attribute is present, we are not considering it as Json indicator, leaving it up to plain Jackson facilities. (if annotation is placed as meta-annotation or for other purposes, the old behavior persist)
released in 2.2.11 Thank you!

**Comment by tetv** [comment]
Thank you so much @elucash.
I have upgraded to 2.2.11 and the generated code seems the same.
I am just using `@JsonDeserialize(builder = ImmutableOrder.Builder.class)`.
What am I doing wrong?

**Comment by elucash** [comment]
you can try it again in 2.2.12, now it should work, hopefully

**Comment by tetv** [comment]
Thank yo @elucsh once again. Now it seems that is working as expected. 👍

**Comment by tetv** [comment]
When we use `Optional<SomeTypeDifferentThanString>` there is a Jackson Error: **IllegalArgumentException: Conflicting setter definitions for property.**

One of the solution is to specify in somehow the generation of only one setter per field in the builder (for example if `optionalAcceptNullable` is set to `true` then the only setter should be the @Nullable one, if is set to `false` then the only setter should be the `Optional` one).

But if there is another way to bypass this error, please let me know.

````java
public abstract class Foo {
    public abstract Optional<Point> maybeBar();
    // ... additional code (Builder, ...)
}

public abstract class Point {
    public abstract int x();
    public abstract int y();
}
````

**Comment by tetv** [comment]
I found I way to bypass the error using `@JsonProperty` in the builder definition:

````java

@Value.Style(
        get = {"get*", "maybe*"},
        builder = "new",
        optionalAcceptNullable = true,
        visibility = Value.Style.ImplementationVisibility.PACKAGE,
        builderVisibility = Value.Style.BuilderVisibility.PACKAGE
)
@Value.Immutable(copy = false)
@JsonDeserialize(builder = ImmutableFoo.Builder.class)
public abstract class Foo {
    public abstract Optional<Point> maybeBar();

    @JsonPOJOBuilder(withPrefix = "set")
    public abstract static class Builder {
        @JsonProperty public abstract Builder setBar(@Nullable Bar bar);
    }
}
````

**Comment by tetv** [comment]
This feature is broken in version 2.3.5.
That means the generated files stat having `@JsonProperty` again.
Note that up to 2.3.4 everything was working fine.

**Comment by elucash** [comment]
I actually enabled it to work with generated builders, there's no `Json` generated, only `@JsonProperty` on immutable getters and builder initializers, there's also `forceJacksonPropertyNames=false` to not output property names if you rely on naming conventions to be auto-applied by Jackson (using strategies etc)
Please, can you explain the actual issue you're facing now?

**Comment by tetv** [comment]
Basically I would like a immutable generated class without all the `@JsonProperty`, because my Jackson is configured to `get` directly from fields (not from properties) and `set` using the `builder` pattern (set properties).

One suggestion is, via `Style`, the option to activate the generation of `@JsonProperty` for the ones that really need it. However it's always good practice to keep the generated files as clean as possible.

**Comment by elucash** [comment]
I find it's useful to be able to set deserialization using generated builder, so everything would work as expected:  https://github.com/immutables/immutables/blob/master/value-fixture/src/org/immutables/fixture/jackson/CustomBuilderDeserialize.java

In this situation, we can go forward with fully disabling any Jackson integration using new special style. My apologies for the inconvenience caused by the change.

**Comment by elucash** [comment]
Added style flag to disable unwanted integration (`jacksonIntegration = false`).
Released in 2.3.7

**Comment by tetv** [comment]
Thank you @elucash, it's working after adding that new style flag.
